### DISCUSSION TITLE
GridGeoSampler for Non-geospatial Data

### DISCUSSION THREAD

**Discussion by Ziyang-cyber** [torchgeo/torchgeo#1157] [General]
May I ask is there any sampler similar to GridGeoSampler for Non-geospatial Data? 
I noticed that you implemented the _RandomNCrop, but it is not reasonable to use it in the validation and testing.

**Comment by adamjstewart** [comment]
@ashnair1 may have some ideas about this. We temporarily had an `_ExtractTensorPatches` method that did what you're looking for, but it was removed in #992 because it wasn't computationally feasible (the resulting batch sizes would be much too large).
